lib/util.ex

```diff
@@ -67,27 +77,33 @@ defmodule Util do
       Authorization: "Bearer " <> auth
     ]
 
-    {:ok, resp} = HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers)
-
-    resp.body
-    |> JSON.decode!()
-    |> Map.get("data")
-    |> Map.get("tournaments")
-    |> Map.get("nodes")
-    # Finesse data into our own non-start.gg format.
-    |> Enum.map(fn sgg_data ->
-      %{
-        "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
-        "name" => sgg_data["name"],
-        "location" => %{
-          "lat" => sgg_data["lat"],
-          "lng" => sgg_data["lng"]
-        },
-        "start_time" => sgg_data["startAt"],
-        "end_time" => sgg_data["endAt"],
-        "updated_at" => sgg_data["updatedAt"],
-        "url" => "https://start.gg/" <> (sgg_data["slug"])
-      }
-    end)
+    case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
+      {:ok, resp} ->
+        try do
+          resp.body
+          |> JSON.decode!
+          |> Map.get("data")
```

Comment:
This tends to be where our currently-ambiguous errors are coming from. We assume that we get data back, and that that data has tournaments and nodes within that. If anything goes wrong (missing auth token, some issue on start.gg's end, who knows), all we see is the error from accessing a nil map.

What we should do instead is some sort of matching on the result of JSON.decode! to make sure that the data has this structure we expect. If it doesn't just printing whatever we did get back to stderr should be enough, since said struct ought to include some sort of debugging info.